.github/workflows/ci.yml

```diff
@@ -72,6 +72,27 @@ jobs:
       run: cargo hack check --all-features --locked --rust-version --ignore-private --workspace --all-targets
     - name: No-default features
       run: cargo hack check --no-default-features --locked --rust-version --ignore-private --workspace --all-targets
+  # Make sure the library builds with all dependencies downgraded to their
+  # oldest versions allowed by the semver spec. This ensures we have not
+  # under-specified any dependency
+  minimal-versions:
+    name: Minimal versions
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    - name: Install stable Rust
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: stable
+    - name: Install nightly Rust
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: stable
+    - name: Downgrade dependencies to minimal versions
+      run: cargo +nightly update -Z minimal-versions
```

Comment:
Since already making a change, could you use `generate-lockfile` rather than `update`?  It'll make the intent clearer.

Reply:
Oh. I did not know `generate-lockfile` could be used for this. Changed as you wish :sparkles:

However IMHO  `update` is more clear. Update is for modifying what version of dependencies the crate uses (`cargo update -Z minimal-versions` is the opposite of `cargo update` where everything is updated to the lowest possible version rather than the highest possible version).

Reply:
For me, `cargo update` implies going up and so it reads weird to see it go down.  `cargo generate-lockfile` implies "throw out the current file and start over again" which is what we want.

Reply:
Ok. Makes no difference in practice. But just fun to discuss. If so, how do you downgrade a single package? `cargo upgrade --precise` is the only way I know of. I know "upgrade" sort of implies it goes upwards. But since it's used to change versions of specific packages both up and down I see it more of as a version-adjustment command.